docs/ml-guide.md

```diff
@@ -40,6 +40,7 @@ Also, some algorithms have additional capabilities in the `spark.ml` API; e.g.,
 provide class probabilities, and linear models provide model summaries.
 
 * [Feature extraction, transformation, and selection](ml-features.html)
+* [Clustering](ml-clustering.html)
```

Comment:
Just noticed that "Feature extraction..." is not alphabetized (sorry about my earlier comment!). If you don't mind, could you move that down under "Ensembles"?